### PR TITLE
Fix 3 packets from issues

### PR DESCRIPTION
**Pull request by MaxtorCoder**
- Fix SMSG_SCENARIO_STATE
- Fix CMSG_CREATE_CHARACTER
- Fix SMSG_WORLD_SERVER_INFO

Fixes: 
- #575 
- #576 
- #579

### CODE REVIEW

**Comment by Aokromes** [comment]
Conflicting files
WowPacketParserModule.V9_0_1_36216/Parsers/CharacterHandler.cs

**Comment by MaxtorCoder** [comment]
> Conflicting files
> WowPacketParserModule.V9_0_1_36216/Parsers/CharacterHandler.cs

Yeah fixing it because of my other PR.

File: WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs [hunk 1]
```diff
@@ -176,5 +176,27 @@ public static void HandlePlayObjectSound(Packet packet)
...
+            packet.ReadByte("IsTournamentRealm");
+
+            packet.ReadBit("XRealmPvpAlert");
+            packet.ReadBit("BlockExitingLoadingScreen");
+            var hasRestrictedAccountMaxLevel = packet.ReadBit("HasRestrictedAccountMaxLevel");
+            var hasRestrictedAccountMaxMoney = packet.ReadBit("HasRestrictedAccountMaxMoney");
+            var hasInstanceGroupSize = packet.ReadBit("HasInstanceGroupSize");
+
+            if (hasRestrictedAccountMaxLevel)
+                packet.ReadInt32("RestrictedAccountMaxLevel");
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs:193]: uint32

File: WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs [hunk 2]
```diff
@@ -176,5 +176,27 @@ public static void HandlePlayObjectSound(Packet packet)
...
+
+            if (hasRestrictedAccountMaxMoney)
+                packet.ReadUInt64("RestrictedAccountMaxMoney");
+
+            if (hasInstanceGroupSize)
+                packet.ReadInt32("InstanceGroupSize");
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs:199]: uint32

File: WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+namespace WowPacketParserModule.V9_0_1_36216.Parsers
+{
+    public static class ScenarioHandler
+    {
+        [Parser(Opcode.SMSG_SCENARIO_STATE)]
+        public static void HandleScenarioState(Packet packet)
+        {
+            packet.ReadInt32("ScenarioID");
+            packet.ReadInt32("CurrentStep");
+            packet.ReadInt32<DifficultyId>("DifficultyID");
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs:14]: All fields between DifficultyID - SpellsCount (also including both) should be uint32

File: WowPacketParserModule.V9_0_1_36216/Parsers/AchievementHandler.cs [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+namespace WowPacketParserModule.V9_0_1_36216.Parsers
+{
+    public static class AchievementHandler
+    {
+        public static Dictionary<int, ulong> FactionReputationStore { get; } = new Dictionary<int, ulong>();
+
+        public static void ReadCriteriaProgress(Packet packet, params object[] indexes)
+        {
+            int criteriaId = packet.ReadInt32<CriteriaId>("CriteriaID", indexes);
+            ulong quantity = (ulong)packet.ReadInt64("Quantity", indexes);
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/AchievementHandler.cs:15]: uint64

**Review by mdX7 — CHANGES_REQUESTED** [review]

File: WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs (hunk 1 above)
Comment (MaxtorCoder) [reply WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs:193]: Was copied from struct in WPP (will change)

File: WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs (hunk 2 above)
Comment (MaxtorCoder) [reply WowPacketParserModule.V9_0_1_36216/Parsers/MiscellaneousHandler.cs:199]: Was copied from struct in WPP (will change)

File: WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs (hunk 3 above)
Comment (MaxtorCoder) [reply WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs:14]: Was copied from struct in WPP (will change)

File: WowPacketParserModule.V9_0_1_36216/Parsers/AchievementHandler.cs (hunk 4 above)
Comment (MaxtorCoder) [reply WowPacketParserModule.V9_0_1_36216/Parsers/AchievementHandler.cs:15]: Was copied from struct in WPP (will change)

File: WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs [hunk 5]
```diff
@@ -9,14 +9,14 @@ public static class ScenarioHandler
...
         {
-            packet.ReadInt32("ScenarioID");
-            packet.ReadInt32("CurrentStep");
-            packet.ReadInt32<DifficultyId>("DifficultyID");
-            packet.ReadInt32("WaveCurrent");
-            packet.ReadInt32("WaveMax");
-            packet.ReadInt32("TimerDuration");
-            var criteriaProgressCount = packet.ReadInt32("CriteriaProgressCount");
-            var bonusObjectiveDataCount = packet.ReadInt32("BonusObjectiveDataCount");
+            packet.ReadUInt32("ScenarioID");
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs:12]: this was actually correct, int32

File: WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs [hunk 6]
```diff
@@ -9,14 +9,14 @@ public static class ScenarioHandler
...
+            packet.ReadUInt32("CurrentStep");
```
Comment (mdX7) [inline WowPacketParserModule.V9_0_1_36216/Parsers/ScenarioHandler.cs:13]: same here

**Review by mdX7 — APPROVED** [review]

**Pull request merged**